pkg/graveler/ref/manager.go

```diff
@@ -412,19 +445,72 @@ func (m *Manager) ListCommits(ctx context.Context, repositoryID graveler.Reposit
 }
 
 func (m *Manager) FillGenerations(ctx context.Context, repositoryID graveler.RepositoryID) error {
```

Comment:
I would extract the code in this function into 2-3 functions for readability and possibly modularity (if some parts will be needed for different use cases).. 
For example a function that builds the tree and a function that executes the BFS is quite reasonable I think